tests/integrationv2/test_sslyze.py

```diff
@@ -0,0 +1,251 @@
+    sslyze.ScanCommand.TLS_1_3_EARLY_DATA,
+    sslyze.ScanCommand.TLS_FALLBACK_SCSV,
+    sslyze.ScanCommand.HEARTBLEED,
+    sslyze.ScanCommand.OPENSSL_CCS_INJECTION,
+    sslyze.ScanCommand.SESSION_RENEGOTIATION
+}
+
+CERTS_TO_TEST = [
+    cert for cert in ALL_TEST_CERTS if cert.name not in {
+        "RSA_PSS_2048_SHA256"  # sslyze doesn't like this cert
+    }
+]
+
+CIPHER_SUITE_SCANS = {
+    Protocols.SSLv3.value: sslyze.ScanCommand.SSL_3_0_CIPHER_SUITES,
+    Protocols.TLS10.value: sslyze.ScanCommand.TLS_1_0_CIPHER_SUITES,
+    Protocols.TLS11.value: sslyze.ScanCommand.TLS_1_1_CIPHER_SUITES,
+    Protocols.TLS12.value: sslyze.ScanCommand.TLS_1_2_CIPHER_SUITES,
+    Protocols.TLS13.value: sslyze.ScanCommand.TLS_1_3_CIPHER_SUITES
+}
+
+
+def get_scan_attempts(scan_results):
+    scan_attribute_names = [attr_name for attr_name in dir(scan_results) if not attr_name.startswith("__")]
+    scan_attempts = [getattr(scan_results, attr_name) for attr_name in scan_attribute_names]
+    scan_attempts = [
+        scan_attempt for scan_attempt in scan_attempts
+        if scan_attempt.status != sslyze.ScanCommandAttemptStatusEnum.NOT_SCHEDULED
+    ]
+    return scan_attempts
+
+
+def assert_scan_result_completed(scan_result):
+    def get_connectivity_error_str(tb):
+        return "\n".join(tb.stack.format())
+
+    assert scan_result.connectivity_status == sslyze.ServerConnectivityStatusEnum.COMPLETED, \
+        f"sslyze could not connect to server: {get_connectivity_error_str(scan_result.connectivity_error_trace)}"
+
+
+def assert_scan_attempt_completed(scan_attempt):
+    assert scan_attempt.status == sslyze.ScanCommandAttemptStatusEnum.COMPLETED, \
+        f"scan attempt ({scan_attempt}) failed: {scan_attempt.status}"
+
+
+def validate_scan_result(scan_attempt, protocol):
+    assert_scan_attempt_completed(scan_attempt)
+
+    scan_result = scan_attempt.result
+    scan_passed = {
+        sslyze.RobotScanResult: {
+            Protocols.TLS13.value:
+                lambda scan:
+                    scan.robot_result == sslyze.RobotScanResultEnum.NOT_VULNERABLE_RSA_NOT_SUPPORTED
+        }.get(
+            protocol.value,
+            lambda scan:
+                scan.robot_result == sslyze.RobotScanResultEnum.NOT_VULNERABLE_NO_ORACLE
+        ),
+        sslyze.SessionResumptionSupportScanResult: {
+            Protocols.TLS13.value: lambda scan: True  # ignore session resumption scan result for tls13
+        }.get(
+            protocol.value,
+            lambda scan:
+                scan.tls_ticket_resumption_result == sslyze.TlsResumptionSupportEnum.FULLY_SUPPORTED
+        ),
+        sslyze.CompressionScanResult:
+            lambda scan: scan.supports_compression is False,
+        sslyze.EarlyDataScanResult: {
+            Protocols.TLS13.value:
+                lambda scan: scan.supports_early_data is True
+        }.get(
+            protocol.value,
+            lambda scan: scan.supports_early_data is False
```

Comment:
For TLS 1.3, scan.supports_early_data is checked to be True. Otherwise, the default value is returned which checks that it's False. So only TLS 1.3 is checked for supporting early data.